src/render/painter.js

```diff
@@ -25,14 +26,66 @@ const draw = {
     debug: require('./draw_debug')
 };
 
+import type Transform from '../geo/transform';
+import type Tile from '../source/tile';
+import type TileCoord from '../source/tile_coord';
+import type {Program} from '../data/program_configuration';
+import type Style from '../style/style';
+import type StyleLayer from '../style/style_layer';
+import type LineAtlas from './line_atlas';
+import type SpriteAtlas from '../symbol/sprite_atlas';
+import type GlyphSource from '../symbol/glyph_source';
+
+type PainterOptions = {
+    showOverdrawInspector: boolean,
+    showTileBoundaries: boolean
+}
+
 /**
  * Initialize a new painter object.
  *
  * @param {Canvas} gl an experimental-webgl drawing context
  * @private
  */
 class Painter {
-    constructor(gl, transform) {
+    gl: WebGLRenderingContext;
+    transform: Transform;
+    _tileTextures: { [number]: Array<WebGLTexture> };
+    frameHistory: FrameHistory;
+    numSublayers: number;
+    depthEpsilon: number;
+    lineWidthRange: [number, number];
+    basicFillProgramConfiguration: ProgramConfiguration;
+    emptyProgramConfiguration: ProgramConfiguration;
+    width: number;
+    height: number;
+    viewportTexture: WebGLTexture;
+    viewportFbo: WebGLFramebuffer;
+    _depthMask: boolean;
+    tileExtentBuffer: Buffer;
+    tileExtentVAO: VertexArrayObject;
+    tileExtentPatternVAO: VertexArrayObject;
+    debugBuffer: Buffer;
+    debugVAO: VertexArrayObject;
+    rasterBoundsBuffer: Buffer;
+    rasterBoundsVAO: VertexArrayObject;
+    extTextureFilterAnisotropic: any;
+    extTextureFilterAnisotropicMax: any;
+    _tileClippingMaskIDs: { [number]: number };
+    style: Style;
+    options: PainterOptions;
+    lineAtlas: LineAtlas;
+    spriteAtlas: SpriteAtlas;
+    glyphSource: GlyphSource;
+    depthRange: number;
+    isOpaquePass: boolean;
+    currentLayer: number;
+    id: string;
+    _showOverdrawInspector: boolean;
+    cache: { [string]: Program };
+    currentProgram: Program;
```

Comment:
For now I'm either listing them in the order that flow complains about, or grouping by functionality. I think the latter is preferable. For classes like `Painter`, we clearly just have way too many members and should refactor to a smaller set.